chapters/memory.adoc

```diff
@@ -3108,18 +3114,174 @@ ifdef::VK_KHR_device_group[]
+function arguments.
+
+include::{generated}/validity/protos/vkMapMemory2EXT.adoc[]
+--
+
+[open,refpage='VkMemoryMapInfoEXT',desc='Structure containing parameters of a memory map operation',type='structs']
+--
+The sname:VkMemoryMapInfoEXT structure is defined as:
+
+include::{generated}/api/structs/VkMemoryMapInfoEXT.adoc[]
+
+  * pname:sType is the type of this structure.
+  * pname:pNext is `NULL` or a pointer to a structure extending this
+    structure.
+  * pname:memory is the slink:VkDeviceMemory object to be mapped.
+  * pname:offset is a zero-based byte offset from the beginning of the
+    memory object.
+  * pname:size is the size of the memory range to map, or
+    ename:VK_WHOLE_SIZE to map from pname:offset to the end of the
+    allocation.
+  * pname:flags is a bitmask of elink:VkMemoryMapFlagBits specifying
+    additional parameters of the memory map operation.
+
+.Valid Usage
+****
+  * pname:memory must: not be currently host mapped
+  * pname:offset must: be less than the size of pname:memory
+  * If pname:size is not equal to ename:VK_WHOLE_SIZE, pname:size must: be
+    greater than `0`
+  * If pname:size is not equal to ename:VK_WHOLE_SIZE, pname:size must: be
+    less than or equal to the size of the pname:memory minus pname:offset
+  * pname:memory must: have been created with a memory type that reports
+    ename:VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT
+ifdef::VK_KHR_device_group[]
+  * pname:memory must: not have been allocated with multiple instances
+endif::VK_KHR_device_group[]
+ifdef::VK_EXT_map_memory_placed[]
+  * If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in pname:flags, the
+    <<features-memoryMapPlaced, pname:memoryMapPlaced>> feature must: be
+    enabled
+  * If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in pname:flags, the
+    pname:pNext chain must: include a slink:VkMemoryMapPlacedInfoEXT
+    structure
+  * If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in pname:flags and the
+    <<features-memoryMapRangePlaced, pname:memoryMapRangePlaced>> feature
+    is not enabled, pname:offset must: be zero
+  * If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in pname:flags and the
+    <<features-memoryMapRangePlaced, pname:memoryMapRangePlaced>> feature
+    is not enabled, pname:size must: be ename:VK_WHOLE_SIZE
+  * If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in pname:flags and the
+    <<features-memoryMapRangePlaced, pname:memoryMapRangePlaced>> feature
+    is enabled, pname:offset must: be aligned to an integer multiple of
+    sname:VkPhysicalDeviceMapMemoryPlacedPropertiesEXT::pname:minPlacedMemoryMapAlignment
+  * If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in pname:flags and the
+    <<features-memoryMapRangePlaced, pname:memoryMapRangePlaced>> feature
+    is enabled, pname:size must: be ename:VK_WHOLE_SIZE or be aligned to an
+    integer multiple of
+    sname:VkPhysicalDeviceMapMemoryPlacedPropertiesEXT::pname:minPlacedMemoryMapAlignment
```

Comment:
`VkPhysicalDeviceLimits::minMemoryMapAlignment` in normal `vkMapMemory` has some other constraint for ppData and offset. Is there any interaction between it and `VkPhysicalDeviceMapMemoryPlacedPropertiesEXT::minPlacedMemoryMapAlignment`?

Reply:
How I read it:

If `minPlacedMemoryMapAlignment` is just as or less granular than `minMemoryMapAlignment`, there is no relevant interaction here, as `*ppData` = `pPlacedAddress`, and both `pPlacedAddress` and `offset` must be aligned to `minPlacedMemoryMapAlignment`.

But yeah, if `minPlacedMemoryMapAlignment` were ever more granular, capitalising on that granularity would be illegal, because `pPlacedAddress` - `offset` still needs to align to `minMemoryMapAlignment`.

I can't imagine a situation where the minimum placed alignment should be more granular than normal, so I'm not sure if this actually a problem.

Reply:
`minMemoryMapAlignment` can be as low as 1 because the driver can fake offsets internally. In fact, Windows drivers have to because the only Windows interface for mapping GPU memory maps the whole memory object every time.  With placed maps, however, you are actually restricted to the OS page size and there's no getting around that.